be/src/io/cache/block_file_cache_downloader.cpp

```diff
@@ -265,10 +265,12 @@ void FileCacheBlockDownloader::download_segment_file(const DownloadFileMeta& met
 
     std::unique_ptr<char[]> buffer(new char[one_single_task_size]);
 
+    size_t task_offset = 0;
     for (size_t i = 0; i < task_num; i++) {
-        size_t offset = meta.offset + i * one_single_task_size;
-        size_t size =
-                std::min(one_single_task_size, static_cast<size_t>(meta.download_size - offset));
+        size_t offset = meta.offset + task_offset;
```

Comment:
add unit test for this function using syncpoint to construct  meta and reader (use local FS & filereader instead)

Reply:
sure, I'd love to.